File: Cs50x 4 Projects/C/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Prototipy of functions, inside main
int count_letters(string words);
int count_words(string words);
int count_sentences(string words);
int calculate_index(int number_of_letters, int number_of_words, int number_of_sentences);

int main(void)
{ // promopt words
    string words = get_string("Text: ");
    // count number of letters
    int number_of_letters = count_letters(words);
    // count number of words
    int number_of_words = count_words(words);
    // count sentenses
    int number_of_sentences = count_sentences(words);
    // Parameter for index
    // Avarage number of letters evry 100 words.
    int index = calculate_index(number_of_letters, number_of_words, number_of_sentences);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string words)
{
    // definition of variables
    int i;
    int N;
    int number_of_letters = 0;
    // definition of leng word
    N = strlen(words);
    // Count of letter using loop for
    for (i = 0; i < N; i++)
        if (isalpha(words[i]))
        {

           number_of_letters += 1;
        }
        else
        {
        }
    // return value in main function
    return number_of_letters;
}

int count_words(string words)
{
    int z;
    int G;
    int number_of_words = 1;
    G = strlen(words);
    // Definition of leng word
    for (z = 0; z < G; z++)
    {
        if (words[z] == ' ')

       number_of_words += 1;
    }
    // return value in main function
    return number_of_words;
}

int count_sentences(string words)
{
    int f;
    int I;
    int number_of_sentences = 0;
    I = strlen(words);
    // Definition of leng word
    for (f = 0; f < I; f++)
    {
        if (words[f] == '.' || words[f] == '!' || words[f] == '?')

       number_of_sentences += 1;
    }
    // return value in main function
    return number_of_sentences;
}

int calculate_index(int number_of_letters, int number_of_words, int number_of_sentences)
{
    float L = (float) number_of_letters / number_of_words * 100;
    float S = (float) number_of_sentences / number_of_words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}

File: CS50x Projects/Python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Ask for an input
    word = input()
    #print(f"Word of input is : {word}")

    # count number of letters
    n_letters = count_letters(word)
    print(f"Number of letters is : {n_letters}")

    # count number of words
    n_words = count_words(word)
    print(f"Number of letters words : {n_words}")

    n_sentences = count_sentences(word)
    print(f"Number of letters sentence : {n_sentences}")

    index = calculate_index(n_letters, n_words, n_sentences)
    #print(f"Index is  : {index}")
    if index < 1:
        print("Before Grade 1")
    elif (index > 16):
        print("Grade 16+")
    else:
        print(f"Grade {index}")











# fuction to have input
def input():
        try:
            word = get_string("Text: ")
            return word
        except ErrorValue:
            print("Error : axept only a string")


def count_letters(word):
    # calculation len
    N = len(word)
    n_letters = 0
    # Iterate through all characters in the word
    n = 0
    while n != N:
        if word[n].isalpha():
            n_letters += 1
        n += 1

    return n_letters


def count_words(word):
      # calculation len
    N = len(word)
    n_words = 1
    # Iterate through all characters in the word
    z = 0
    while z != N:
        if word[z] == " ":
            n_words += 1
        z += 1

    return n_words


def count_sentences(word):
      # calculation len
    N = len(word)
    n_sentences = 0
    # Iterate through all characters in the word
    r = 0
    while r != N:
        if word[r] == "." or word[r] == "!" or word[r] == "?" :
            n_sentences += 1
        r += 1

    return n_sentences



def calculate_index(n_letters, n_words, n_sentences):
    L = n_letters / n_words * 100
    S = n_sentences / n_words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index











init = "__main__"
main()

